src/generic_ui/polymer/contact.ts

```diff
@@ -111,10 +115,74 @@ Polymer({
       },
       userId: this.contact.userId // Cloud instance userId
     }).then((cloudInviteUrl: string) => {
-      ui.showDialog(ui.i18n_t("CLOUD_SHARE_INSTRUCTIONS"), '', ui.i18n_t("OK"),
+      this.ui.showDialog(this.ui.i18n_t("CLOUD_SHARE_INSTRUCTIONS"), '', this.ui.i18n_t("OK"),
         undefined, cloudInviteUrl);
     });
   },
+  displayCloudRemovalConfirmation: function() {
+    // We keep track of which contact should be deleted with toBeDeleted
+    // TODO: Instead of using toBeDeleted, send the contact's ID as data
+    // with a core-signal
+    this.toBeDeleted = true;
+    // Core signals for the confirmation dialog cancel and continue buttons
+    var coreSignals = {
+      'cancel': 'cancel-cloud-removal',
+      'continue': 'remove-cloud-server'
+    };
+    if (this.contact.status === this.UserStatus.CLOUD_INSTANCE_CREATED_BY_LOCAL) {
+      this.ui.getConfirmation(this.ui.i18n_t("REMOVE_CLOUD_SERVER"), this.ui.i18n_t('DESTROY_CLOUD_SERVER_CONFIRMATION'),
+        true, coreSignals);
+    } else {
+      this.ui.getConfirmation(this.ui.i18n_t("REMOVE_CLOUD_SERVER"), this.ui.i18n_t('REMOVE_CLOUD_SERVER_CONFIRMATION'),
+        true, coreSignals);
+    }
+  },
+  cancelCloudRemoval: function() {
+    // toBeDeleted should be false if the user cancels cloud removal
+    this.toBeDeleted = false;
+  },
+  removeCloudFriend: function() {
+    // Don't do anything if this contact is not the one to be deleted
+    if (!this.toBeDeleted) {
+      return;
+    }
+    this.toBeDeleted = false;
+    // Destroy cloud server if created by user
+    this.destroyCloudServerIfNeeded().then(() => {
+      // Remove contact from friend list
+      ui_context.core.removeContact({
+        networkName: this.contact.network.name,
+        userId: this.contact.userId
+      });
+    }).then(() => {
+      this.ui.showDialog(this.ui.i18n_t("REMOVE_CLOUD_SERVER"), this.ui.i18n_t("REMOVE_CLOUD_SERVER_SUCCESS"),
+        this.ui.i18n_t("OK"), undefined, undefined);
+    }).catch((e: Error) => {
+      if (e.name === 'CLOUD_ERR') {
+         this.ui.showDialog(this.ui.i18n_t("REMOVE_CLOUD_SERVER"), this.ui.i18n_t("DESTROY_CLOUD_SERVER_FAILURE"),
+          this.ui.i18n_t("OK"), undefined, undefined);
```

Comment:
These last 3 params to ui.showDialog just match the default values, might be a little more readable to remove them